docs/index.rst

```diff
@@ -1,260 +1,427 @@
-this method to give you a directory that actually exists.
+::
 
-As an example, you may want to migrate a user's settings to Confit from
-an older configuration system such as `ConfigParser`_. Just do something
-like this::
+  as_str(regex)
 
-    config_filename = os.path.join(config.config_dir(),
-                                   confit.CONFIG_FILENAME)
-    with open(config_filename, 'w') as f:
-        yaml.dump(migrated_config, f)
+Return a string if it matches the regular expression.
 
-.. _ConfigParser: http://docs.python.org/library/configparser.html
+::
 
+  as_strs(split=str.split)
 
-Dynamic Updates
----------------
+Given either a string or a list of strings, return a list of strings. A
+single string is split using *split*; the default splits on whitespace.
 
-Occasionally, a program will need to modify its configuration while it's
-running. For example, an interactive prompt from the user might cause
-the program to change a setting for the current execution only. Or the
-program might need to add a *derived* configuration value that the user
-doesn't specify.
+::
 
-To facilitate this, Confit lets you *assign* to view objects using
-ordinary Python assignment. Assignment will add an overlay source that
-precedes all other configuration sources in priority. Here's an example
-of programmatically setting a configuration value based on a ``DEBUG``
-constant::
+  as_date(formats)
 
-    if DEBUG:
-        config['verbosity'] = 100
-    ...
-    my_logger.setLevel(config['verbosity'].get(int))
+::
 
-This example allows the constant to override the default verbosity
-level, which would otherwise come from a configuration file.
+  as_time(formats)
 
-Assignment works be creating a new "source" for configuration data at
-the top of the stack. This new source takes priority over all other,
-previously-loaded sources. You can cause this explicitly by calling the
-``set()`` method on any view. A related method, ``add()``, works
-similarly but instead adds a new *lowest-priority* source to the bottom
-of the stack. This can be used to provide defaults for options that may
-be overridden by previously-loaded configuration files.
+::
+
+  as_datetime(formats)
+
+::
+
+  as_filename
+
+Return a filename, substituting tildes and absolute-ifying relative paths.
+The filename is relative to the source that provided it. That is, a relative
+path in a file (including a Python module) is relative to the directory
+containing the file. A relative path in a command-line argument or an
+environment variable is relative to the current working directory.
+
+::
+
+  as_choice(choices)
+
+Assert the value is among the *choices* (using the ``in`` operator) and, if
+*choices* is a ``dict``, return the associated value.
+
+
+Custom casts
+^^^^^^^^^^^^
+
+It is easy to define your own casts. A cast is just a function that takes a
+stack of variables, in precedence order starting with most preferred, and
+returns a value or raises an exception.
```

Comment:
Perhaps a simpler interface would just pass the `Variable` object itself to the function (rather than `var.stack()`). It could then choose whether to get the stack, be sensitive to the origin, or other such details.